torch/distributed/distributed_c10d.py

```diff
@@ -2429,6 +2429,47 @@ def barrier(group=GroupMember.WORLD,
     else:
         work.wait()
 
+def monitored_barrier(group=GroupMember.WORLD, timeout=None):
+
+    """
+    Synchronizes all processes similar to torch.distributed.barrier, but takes
+    a configurable timeout and is able to report ranks that did not pass this
+    barrier within that timeout. If the rank is part of the passed in group, it
+    will not return from monitored_barrier until all processes have successfully
+    synchronized, if not part of the group, will return immediately.
+
+    This collective always blocks processes until the whole group exits this
+    function successfully. Therefore, it can have a performance penalty and
+    should only be used for debugging or scenarios that require full
+    synchronization points on the host-side.
+
+    .. note:: Note that this collective is only supported with the GLOO backend.
+
+    Args:
+        group (ProcessGroup, optional): The process group to work on. If None,
+            the default process group will be used.
+        timeout (float, optional): Timeout in seconds for monitored_barrier.
```

Comment:
We use `timedelta` for timeout in other APIs like `init_process_group`, should we use `timedelta` here too?

Reply:
I see, main reason I did float was for torchscript compatibility and that we did it for RPC framework, but let me check if we can do timedelta and it works with torchscript...